hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveCatalog.java

```diff
@@ -358,6 +388,36 @@ public void testCreateNamespace() throws Exception {
         "There no same location for db and namespace", database2.getLocationUri(), hiveLocalDir);
   }
 
+  @Test
+  public void testCreateNamespaceWithOwnership() throws Exception {
+    Namespace namespace1 = Namespace.of("userOwnership");
+    Map<String, String> prop1 =
+        ImmutableMap.of(
+            TableProperties.HMS_DB_OWNER,
+            "apache",
+            TableProperties.HMS_DB_OWNER_TYPE,
+            PrincipalType.USER.name());
+    catalog.createNamespace(namespace1, prop1);
+    Database database1 = metastoreClient.getDatabase(namespace1.toString());
+
+    Assert.assertEquals("apache", database1.getOwnerName());
+    Assert.assertEquals(PrincipalType.USER, database1.getOwnerType());
+
+    Map<String, String> prop2 =
+        ImmutableMap.of(
+            TableProperties.HMS_DB_OWNER,
+            "iceberg",
+            TableProperties.HMS_DB_OWNER_TYPE,
+            PrincipalType.GROUP.name());
+    Namespace namespace2 = Namespace.of("groupOwnership");
+
+    catalog.createNamespace(namespace2, prop2);
+    Database database2 = metastoreClient.getDatabase(namespace2.toString());
+
+    Assert.assertEquals("iceberg", database2.getOwnerName());
+    Assert.assertEquals(PrincipalType.GROUP, database2.getOwnerType());
+  }
```

Comment:
Can we also add a check to verify that PrincipalType.USER is the default when we don't provide the ownership type in parameters?

Reply:
Yes, good idea.